README.md

```diff
@@ -1,13 +1,104 @@
-## My Project
+# Amazon Ion Hash .NET
 
-TODO: Fill this README out!
+An implementation of [Amazon Ion Hash](http://amzn.github.io/ion-hash) in C#.
 
-Be sure to:
+[![Build Status](https://travis-ci.com/amzn/ion-hash-dotnet.svg?branch=master)](https://travis-ci.com/amzn/ion-hash-dotnet)
+[![nuget version](https://img.shields.io/nuget/v/ion-hash-dotnet.svg)](https://www.nuget.org/packages/IonHashDotnet/)
+[![license](https://img.shields.io/hexpm/l/plug.svg)](https://github.com/amzn/ion-hash-dotnet/blob/master/LICENSE)
+[![docs](https://img.shields.io/badge/docs-api-green.svg?style=flat-square)](https://amzn.github.io/ion-hash-dotnet/api)
+
+## Getting Started
+
+This library is designed to work with .NET Core 2.1.  The following example code
+illustrates how to use it:
+
+```C#
+class HashDuringWriteAndRead
+{
+    static void Main(string[] args)
+    {
+        IIonHasherProvider hasherProvider = new CryptoIonHasherProvider("SHA-256");
+
+        // write a simple Ion struct and compute the hash
+        TextWriter tw = new StringWriter();
+        IIonWriter writer = IonTextWriterBuilder.Build(tw);
+        IIonHashWriter hashWriter = IonHashWriterBuilder.Standard()
+            .WithHasherProvider(hasherProvider)
+            .WithWriter(writer)
+            .Build();
+
+        Console.WriteLine("writer");
+        hashWriter.StepIn(IonType.Struct);
+        hashWriter.SetFieldName("first_name");
+        hashWriter.WriteString("Amanda");
+        hashWriter.SetFieldName("middle_name");
+        hashWriter.WriteString("Amanda");
```

Comment:
Did you intend for "Amanda" to be both the first and middle name in the example?

Reply:
Was following my example from https://github.com/amzn/ion-hash-java#example-usage; not sure why I used "Amanda" as the value for both the first and middle name fields.

Note also that this example is different than that shown for https://github.com/amzn/ion-hash-js#getting-started and https://github.com/amzn/ion-hash-python#getting-started.

Perhaps it'd be better to align with the ion-hash-js example and avoid others wondering about this same question?